storage/src/adb/any.rs

```diff
@@ -283,8 +299,19 @@ impl<E: RStorage + Clock + Metrics, K: Array, V: Array, H: CHasher, T: Translato
         value: Option<&V>,
         new_loc: u64,
     ) -> Result<UpdateResult, Error> {
-        let mut loc_iter = snapshot.update_iter(&key);
-        for loc in &mut loc_iter {
+        // Determine if there are any conflicting operations for the key in the snapshot.
+        let Some(mut cursor) = snapshot.get_mut(&key) else {
+            // The key isn't in the snapshot, so add it.
+            snapshot.insert(&key, new_loc);
+            return Ok(UpdateResult::Inserted(new_loc));
+        };
+
+        // Iterate over conflicts in the snapshot.
+        loop {
+            let Some(loc) = cursor.next() else {
+                // The key isn't in the snapshot, so add it.
```

Comment:
a little weird seeing the same comment here as on L326.  Maybe change it to something like "break out of the loop to add key that isn't in snapshot" ?

Reply:
I see the same comment on L304 too which perhaps could use some differentiation.